### PR TITLE
8 packages from savonet/ocaml-ffmpeg at 1.0.0-beta1

### PR DESCRIPTION
**Pull request by toots**
This pull-request concerns:
-`ffmpeg.1.0.0-beta1`: Bindings for the ffmpeg libraries
-`ffmpeg-av.1.0.0-beta1`: Bindings for the ffmpeg libraries -- top-level helpers
-`ffmpeg-avcodec.1.0.0-beta1`: Bindings for the ffmpeg avcodec library
-`ffmpeg-avdevice.1.0.0-beta1`: Bindings for the ffmpeg avdevice library
-`ffmpeg-avfilter.1.0.0-beta1`: Bindings for the ffmpeg avfilter library
-`ffmpeg-avutil.1.0.0-beta1`: Bindings for the ffmpeg avutil libraries
-`ffmpeg-swresample.1.0.0-beta1`: Bindings for the ffmpeg swresample library
-`ffmpeg-swscale.1.0.0-beta1`: Bindings for the ffmpeg swscale library

---
* Homepage: https://github.com/savonet/ocaml-ffmpeg
* Source repo: git+https://github.com/savonet/ocaml-ffmpeg.git
* Bug tracker: https://github.com/savonet/ocaml-ffmpeg/issues

---
:camel: Pull-request generated by opam-publish v2.0.3

### CODE REVIEW

**Comment by toots** [comment]
I see the following errors:
* depext package not present
* `mastodon` failing to build, not sure if it is related
* Some random errors during tests, mostly due to change in ffmpeg's build configuration and an issue with hardware acceleration on ubuntu that could be hardware specific

Otherwise, build and tests work in most situations.

Overall, these errors should be fine to ignore for now I believe.

**Comment by kit-ty-kate** [comment]
I've change the version number from `1.0.0-beta1` to `1.0.0~beta1` per opam's version ordering (see https://opam.ocaml.org/doc/Manual.html#version-ordering), as `1.0.0~beta1 < 1.0.0 < 1.0.0-beta1`

**Comment by toots** [comment]
Thanks for these improvements. I have backported them to `ocaml-ffmpeg`, except for the `pinned` changes, this code is generated by `dune`..

File: packages/ffmpeg-av/ffmpeg-av.1.0.0~beta1/opam [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+authors: ["The Savonet Team <savonet-users@lists.sourceforge.net>"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/savonet/ocaml-ffmpeg"
+bug-reports: "https://github.com/savonet/ocaml-ffmpeg/issues"
+depends: [
+  "conf-pkg-config"
+  "conf-ffmpeg"
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
+  "dune-configurator"
```
Comment (toots) [inline packages/ffmpeg-av/ffmpeg-av.1.0.0~beta1/opam:13]:
```suggestion
  "conf-pkg-config" {build}
  "conf-ffmpeg" {build}
  "ocaml" {>= "4.08.0"}
  "dune" {>= "2.0"}
  "dune-configurator" {build}
```

File: packages/ffmpeg-avdevice/ffmpeg-avdevice.1.0.0~beta1/opam [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+authors: ["The Savonet Team <savonet-users@lists.sourceforge.net>"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/savonet/ocaml-ffmpeg"
+bug-reports: "https://github.com/savonet/ocaml-ffmpeg/issues"
+depends: [
+  "conf-pkg-config"
+  "conf-ffmpeg"
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
+  "dune-configurator"
```
Comment (toots) [inline packages/ffmpeg-avdevice/ffmpeg-avdevice.1.0.0~beta1/opam:13]:
```suggestion
  "conf-pkg-config" {build}
  "conf-ffmpeg" {build}
  "ocaml" {>= "4.08.0"}
  "dune" {>= "2.0"}
  "dune-configurator" {build}
```

File: packages/ffmpeg-avfilter/ffmpeg-avfilter.1.0.0~beta1/opam [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+authors: ["The Savonet Team <savonet-users@lists.sourceforge.net>"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/savonet/ocaml-ffmpeg"
+bug-reports: "https://github.com/savonet/ocaml-ffmpeg/issues"
+depends: [
+  "conf-pkg-config"
+  "conf-ffmpeg"
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
+  "dune-configurator"
```
Comment (toots) [inline packages/ffmpeg-avfilter/ffmpeg-avfilter.1.0.0~beta1/opam:13]:
```suggestion
  "conf-pkg-config" {build}
  "conf-ffmpeg" {build}
  "ocaml" {>= "4.08.0"}
  "dune" {>= "2.0"}
  "dune-configurator" {build}
```

File: packages/ffmpeg-avutil/ffmpeg-avutil.1.0.0~beta1/opam [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+authors: ["The Savonet Team <savonet-users@lists.sourceforge.net>"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/savonet/ocaml-ffmpeg"
+bug-reports: "https://github.com/savonet/ocaml-ffmpeg/issues"
+depends: [
+  "conf-pkg-config"
+  "conf-ffmpeg"
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
+  "dune-configurator"
```
Comment (toots) [inline packages/ffmpeg-avutil/ffmpeg-avutil.1.0.0~beta1/opam:13]:
```suggestion
  "conf-pkg-config" {build}
  "conf-ffmpeg" {build}
  "ocaml" {>= "4.08.0"}
  "dune" {>= "2.0"}
  "dune-configurator" {build}
```

File: packages/ffmpeg-swresample/ffmpeg-swresample.1.0.0~beta1/opam [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+authors: ["The Savonet Team <savonet-users@lists.sourceforge.net>"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/savonet/ocaml-ffmpeg"
+bug-reports: "https://github.com/savonet/ocaml-ffmpeg/issues"
+depends: [
+  "conf-pkg-config"
+  "conf-ffmpeg"
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
+  "dune-configurator"
```
Comment (toots) [inline packages/ffmpeg-swresample/ffmpeg-swresample.1.0.0~beta1/opam:13]:
```suggestion
  "conf-pkg-config" {build}
  "conf-ffmpeg" {build}
  "ocaml" {>= "4.08.0"}
  "dune" {>= "2.0"}
  "dune-configurator" {build}
```

File: packages/ffmpeg-swscale/ffmpeg-swscale.1.0.0~beta1/opam [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+authors: ["The Savonet Team <savonet-users@lists.sourceforge.net>"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/savonet/ocaml-ffmpeg"
+bug-reports: "https://github.com/savonet/ocaml-ffmpeg/issues"
+depends: [
+  "conf-pkg-config"
+  "conf-ffmpeg"
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0"}
+  "dune-configurator"
```
Comment (toots) [inline packages/ffmpeg-swscale/ffmpeg-swscale.1.0.0~beta1/opam:13]:
```suggestion
  "conf-pkg-config" {build}
  "conf-ffmpeg" {build}
  "ocaml" {>= "4.08.0"}
  "dune" {>= "2.0"}
  "dune-configurator" {build}
```

**Comment by kit-ty-kate** [comment]
> Thanks for these improvements. I have backported them to `ocaml-ffmpeg`, except for the `pinned` changes, this code is generated by `dune`..

This has been fixed in dune 2.7.0 but you need to upgrade to `(lang dune 2.7)` or above for dune to actually change the files

**Comment by toots** [comment]
Done, thanks!

**Comment by toots** [comment]
Hi! Any update on this one? I might need to revert the changes on `ocaml-ffmpeg` since it is currently broken without `conf-ffmpeg`..

**Comment by toots** [comment]
> Looks fine to me overall now. Is the Ubuntu error
> 
> ```
> 
> - (cd _build/default/test && ../examples/hw_encode.exe nvenc.mp4 h264_nvenc device)
> - hw_config for h264_nvenc:
> - {
> -   pixel_format: none,
> -   methods: hw_device_ctx,
> -   device_type: cuda
> - },
> - {
> -   pixel_format: cuda,
> -   methods: hw_frames_ctx,
> -   device_type: cuda
> - }
> - Trying device context method..
> - failed with error: Error number 1313558101 occurred
> - Fatal error: exception FFmpeg.Avutils.Error(Unknown error occurred)
> - Raised by primitive operation at Avutil.HwContext.create_device_context in file "avutil/avutil.ml", line 688, characters 6-67
> - Called from Dune__exe__Hw_encode in file "examples/hw_encode.ml", line 162, characters 12-62
> [ERROR] The compilation of ffmpeg failed at "/home/opam/.opam/opam-init/hooks/sandbox.sh build dune build -p ffmpeg -j 127 @install @runtest".
> ```
> 
> something to worry about?

From what I can see, `ffmpeg` detects supports for hardware acceleration but fails to use it. I don't think it's a major issue and could even be a bug with `ffmpeg` itself.

**Comment by mseri** [comment]
Ok, then it is good to merge. Sorry for the delay

**Comment by toots** [comment]
Thanks!

**Pull request merged**